### ISSUE TITLE
NERC OpenShift Rebuild

### ISSUE THREAD

**Issue by joachimweyl** [OCP-on-NERC/operations#51]
- [x] NERC OpenShift Production Rebuild
- [x] NERC OpenShift Production Testing
- [x] NERC OpenShift Infrastructure Rebuild
- [x] NERC OpenShift Infrastructure Testing

**Comment by msdisme** [comment]
<img width="1216" alt="image" src="https://user-images.githubusercontent.com/6126065/210609963-04354810-d96e-4803-9fb1-f955e3f3e645.png">

**Comment by jtriley** [comment]
We've done an initial rebuild of the infra cluster with @larsks and took notes along the way. Next steps are to formalize our notes into a doc, have folks review the steps and correct any issues or add improvements as needed, and then try the rebuild again with just myself and @aabaris to see if we've accounted for everything.

**Comment by larsks** [comment]
In order to re-import the prod cluster to ACM, we had to remove the existing ACM related resources from prod. This would have happened automatically if we had *detached* the prod cluster before re-installing the infra cluster.

The docs for ACM 2.2 have a [section on manually detaching a cluster](https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_management_for_kubernetes/2.2/html/manage_cluster/remove-managed-cluster#removing-a-cluster-from-management-in-special-cases), which points the reader to [this script](https://github.com/stolostron/deploy/blob/master/hack/cleanup-managed-cluster.sh).

I didn't run the script, but I did use it as a model. I started by removing the `klusterlet`:

```
kubectl delete klusterlet klusterlet
```

It was not necessary to explicitly remove the `finalizers` from the resource.

Once the klusterlet was deleted, I removed the CRDs:

```
kubectl get crd -o name |grep open-cluster-management | xargs delete
```

This seemed to be sufficient, and I was able to re-run the import command provided by ACM.

Re-importing the prod cluster to ACM also re-added the cluster to ArgoCD:

```
$ kubectl -n openshift-gitops get secret -l argocd.argoproj.io/secret-type=cluster
NAME                                        TYPE     DATA   AGE
cluster-kubernetes.default.svc-3396314289   Opaque   3      6h4m
nerc-ocp-prod-cluster-secret                Opaque   3      47m
```

But at the moment this has not caused the AoA applicationset to generate the necessary app for the prod cluster.
